src/main-client.ts

```diff
@@ -782,9 +782,28 @@ export class MainClient extends BaseRestClient {
   }
 
   getSymbolPriceTicker(
-    params?: Partial<BasicSymbolParam>,
+    params: BasicSymbolParam,
+  ): Promise<SymbolPrice>;
+
+  getSymbolPriceTicker(
+    params?: SymbolArrayParam,
+  ): Promise<SymbolPrice[]>;
+
+  getSymbolPriceTicker(
+    params?: Partial<BasicSymbolParam> | Partial<SymbolArrayParam>,
   ): Promise<SymbolPrice | SymbolPrice[]> {
-    return this.get('api/v3/ticker/price', params);
+    if (params && typeof params['symbol'] === 'string') {
+      return this.get('api/v3/ticker/price', params);
+    }
+
+    if (params && params['symbols'] && Array.isArray(params['symbols'])) {
+      const symbols = (params as SymbolArrayParam).symbols;
+      const symbolsQueryParam = JSON.stringify(symbols);
+
+      return this.get('api/v3/ticker/price?symbols=' + symbolsQueryParam);
+    }
```

Comment:
GET requests are automatically mapped to a query string, does this not work if you simply pass the object too - similar to your change at line 795?
```
      return this.get('api/v3/ticker/price', params);
```

Nothing wrong with what you did, just checking in case you haven't tried it.

Reply:
I tried it, you really have to put json array string inside the query string here just like in `api/v3/ticker/24hr`.

You can check this on the master branch:

```
$ node -e "Object.assign(global, require('.')); mc = new MainClient(); [x => x, JSON.stringify].map(f => mc.getSymbolPriceTicker({ symbols: f(['BTCUSDT', 'ETHUSDT']) }).then(console.log, console.error));"
 
[
  { symbol: 'BTCUSDT', price: '42954.95000000' },
  { symbol: 'ETHUSDT', price: '2305.66000000' }
]
{
  code: -1101,
  message: "Duplicate values for parameter 'symbols[]'.",
  body: { code: -1101, msg: "Duplicate values for parameter 'symbols[]'." },
  headers: Object [AxiosHeaders] {
    'content-type': 'application/json;charset=UTF-8',
    'content-length': '66',
    connection: 'keep-alive',
    date: 'Sun, 04 Feb 2024 17:13:47 GMT',
    server: 'nginx',
    'x-mbx-uuid': 'a75326aa-3fcb-4788-bc8f-d097b07719a0',
    'x-mbx-used-weight': '10',
    'x-mbx-used-weight-1m': '10',
    'strict-transport-security': 'max-age=31536000; includeSubdomains',
    'x-frame-options': 'SAMEORIGIN',
    'x-xss-protection': '1; mode=block',
    'x-content-type-options': 'nosniff',
    'content-security-policy': "default-src 'self'",
    'x-content-security-policy': "default-src 'self'",
    'x-webkit-csp': "default-src 'self'",
    'cache-control': 'no-cache, no-store, must-revalidate',
    pragma: 'no-cache',
    expires: '0',
    'x-cache': 'Error from cloudfront',
    via: '1.1 4fa064f65088b74bd9abffd69e1e9de4.cloudfront.net (CloudFront)',
    'x-amz-cf-pop': 'SOF50-P2',
    'x-amz-cf-id': 'T4Lbxi-PETLyhXlrsf1xUYwS30xrWsVLxFsoWHmfl7saV8Rt_9Ok5g=='
  },
  requestUrl: 'https://api.binance.com/api/v3/ticker/price',
  requestBody: undefined,
  requestOptions: {
    recvWindow: 5000,
    syncIntervalMs: 3600000,
    strictParamValidation: false,
    disableTimeSync: true,
    api_key: undefined,
    api_secret: undefined
  }
}

```